src/Microsoft.Identity.Client/UiRequiredExceptionClassification.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Details about the cause of an <see cref="MsalUiRequiredException"/>, giving a hint about what the user can expect when 
+    /// they go through interactive authentication. See https://aka.ms/msal-net-UiRequiredException for details.
+    /// </summary>
+    public enum UiRequiredExceptionClassification
+    {
+        /// <summary>
+        /// No further details are provided. It is possible that the user will be able to resolve the issue by launching interactive authentication.
+        /// See https://aka.ms/msal-net-UiRequiredExceptionfor details
+        /// </summary>
+        None,
```

Comment:
MSAL throw UiRequiredException is also thrown when the user calls AcquireTokenSilent with: 

- account = null
- a login_hint that cannot be matched to 1 (and only 1) account

This is breaks the pattern that all client side exceptions are MsalClientException, in order to provide a more uniform experience (most users don't care why the UiRequiredException is thrown). It's also a downside for controlling the flow using exceptions. 

I think we can improve the user experience if we added more options to this enum for the "client" cases, something like "InvalidAccount"